### ISSUE TITLE
<...>.ForEach does not work in a particulaer case

### ISSUE THREAD

**Issue by HeavenMaster** [bridgedotnet/Bridge#1040]
If we use lambda like x => x.BoolProperty = false, ForEach stops after the first item.

### Steps To Reproduce

```csharp
public class App
{
    [Ready]
    public static void Main()
    {
          // Something like below pseudocode
          SomeObject[] array = .....;
          bool on = false;
          array.ForEach(a=> a.IsEnabled = on); // <--- will be invoked for the first item only
    }
}
```

**Comment by HeavenMaster** [comment]
I just note that ForEach has overload with Func<T, bool>, so now I think it is by design, but it is really fonfuses. Possibly API should be revised

**Comment by DaniilVeriga** [comment]
Hi @HeavenMaster,

Thank you for the report! But I could not reproduce it. Here is a [Live Test Case](http://live.bridge.net/#5a4ad86db230d49f06e8).

There are two iterations and the second SomeObject's IsEnabled is actually changed to `false`.

**Comment by HeavenMaster** [comment]
I have modified live test case to make repro http://live.bridge.net/#ad8889195b2787d1823c
Not an real issue, just a little confused me.

**Comment by HeavenMaster** [comment]
BTW, cool live compiler and cool product

**Comment by DaniilVeriga** [comment]
Thank you for the kind words!

As for the issue, I believe I tried with
`array.ForEach(a => a.IsEnabled = on);`
and could not reproduce the issue. But, apparently, I tested somehow wrong.

Your test case reproduces the issue and I think it is a bug. I slightly modified a test case to demonstrate in an obvious way that the first array's item is being changed, but the second is not.

[Live Test Case](http://live.bridge.net/#1c4d7abc377da8751726)

Actual: `false true` in the alert box
Expected: `false false` in the alert box

We will investigate. Thank you for the report!
